### PR TITLE
Replace implicit dependence on ext/iconv in JS escaper

### PR DESCRIPTION
**Pull request by JoshyPHP**
Twig's JS escaper fails on non-word characters if `ext/iconv` isn't available. That's because it uses iconv to replace characters with their hex codepoints and split characters outside the Basic Multilingual Plane into surrogate pairs. `symfony/polyfill-iconv` does not support the UTF-16BE conversion and cannot be used as a replacement.

This PR splits the surrogates using native PHP, after obtaining the codepoint via `mb_ord()` which is provided either natively or via `symfony/polyfill-mbstring`, which is already part of Twig's requirements.

### CODE REVIEW

**Review by nicolas-grekas — APPROVED** [review]
Looks good to me.
Can you please check if/confirm that both cases are covered by a unit test?

**Comment by JoshyPHP** [comment]
You're right, I should have addressed testing. I did look into the current test suite and there already are tests covering non-ASCII characters and surrogates, that's why I didn't add any. I manually confirmed that a non-functional implementation would cause those tests to fail.

https://github.com/twigphp/Twig/blob/b48bd18dc6f967ad09af9eab521cdf0e68fb6a95/tests/Extension/EscaperTest.php#L72-L74

Now, a regression test would need to run with `ext/iconv` disabled and to be thorough the new implementation should be tested with `mbstring` disabled. Unfortunately, I don't think there's a simple way to do that in Travis. Also, unrelated parts of the test suite fail if `ext/iconv` is disabled.

In summary, there's already a couple of test cases for that portion of the code and I manually ran them with both `ext/iconv` and `mbstring` disabled, successfully.

**Comment by fabpot** [comment]
Thank you @JoshyPHP.
